### PR TITLE
Add JS_ArrayGetLength and JS_FreePropertyEnum for external iterators

### PR DESCRIPTION
**Pull request by Icemic**
See #408

### CODE REVIEW

File: quickjs.c [hunk 1]
```diff
@@ -7007,6 +7007,11 @@ static JSValue JS_GetPrototypeFree(JSContext *ctx, JSValue obj)
     return obj1;
 }
 
+/* works for arrays, strings, functions and arguments, but not for typed arrays */
+int JS_ArrayGetLength(JSContext *ctx, int64_t *pres, JSValueConst obj) {
+    return js_get_length64(ctx, pres, obj);
+}
+
```
Comment (chqrlie) [inline quickjs.c]: It is misleading for this function to **not** handle typed arrays given the name.
The name is even more confusing as it does handle strings and functions.

**Comment by saghul** [comment]
Can you please split this into 2 commits at least? We can keep 1 PR, but the 2 are unrelated.

**Comment by Icemic** [comment]
> Can you please split this into 2 commits at least? We can keep 1 PR, but the 2 are unrelated.

Of course, done

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
@chqrlie can you take a quick look at?

**Comment by chqrlie** [comment]
> @chqrlie can you take a quick look at?

Sometime tonight.

**Pull request merged**